.travis.yml

```diff
@@ -43,18 +43,15 @@ script:
   else
     set -e
     ./node_modules/.bin/gulp
-    cd website
-    yarn install
-    ./setup.sh
     if [ "$TRAVIS_PULL_REQUEST" = false ] && [ "$TRAVIS_BRANCH" = master ]; then
       # Automatically publish the website
       git config --global user.name "Travis CI"
       git config --global user.email "travis@reactjs.org"
-      echo "machine github.com login reactjs-bot password $GITHUB_TOKEN" >~/.netrc
-      ./publish.sh
+      echo "machine github.com login reactjs-bot password $GITHUB_TOKEN" > ~/.netrc
+      cd website && yarn install && GIT_USER=reactjs-bot npm run publish-gh-pages
```

Comment:
It's unfortunate that we need these hacks (yes, checking in committed code to Git is a hack). I'd like to eventually move the site over to Netlify to avoid it. That'd also give us preview versions of the site for each PR submitted.

Reply:
Docusaurus uses Netlify to show PR previews. It's pretty easy to set up - https://docusaurus.io/docs/en/1.0.14/publishing.html#hosting-on-netlify

Reply:
@yangshun Yeah it'd be good to have the DraftJS site totally on Netlify, rather than using both Netlify and GitHub Pages. We shouldn't use Netlify for previews but GitHub Pages for the live site, as the difference between the environments could cause issues. Moving the site across would just need some small changes, like moving the DNS over.

Reply:
Would it cost a lot to use Netlify to host the production site though? If it's free it would be great for Docusaurus to do that too (I think docusaurus.io is also using gh-pages now)

Reply:
Fre for open-source :) yarnpkg.com and reactjs.org use it.

Reply:
Awesome! @JoelMarcey WDYT?

Reply:
The primary goal of Docusaurus was to make getting sites up and running easily. GitHub/GitHub Pages seemed to be the easiest way to get folks started on that given that they are used heavily and are integrated.

That said, we know that people want to use Netlify for hosting and CircleCI for integration (we should probably document Travis at some point 😉 ) -- which is why we have documented those.

So it's great if Draft.js wants to move to Netlify exclusively. Others will choose different options as well.

Whether Docusaurus should move exclusively to Netlify, I am not convinced. I think us hosting on GitHub Pages adheres to the first goals of Docusaurus and since we use Netlify to preview, we are showcasing other possibilities too.

Not sure I feel too strongly, but as of this point, I would say I wouldn't actively pursue a move to Netlify. But, I can ask others on the Docusaurus team what they think too.

Reply:
Thanks everyone~!

:) Happy to hear that we should be able to merge this - @NoamELB do you want to do the honors? Or would you prefer I find a time to do it?

Reply:
You can do it 😊